src/phpDocumentor/Descriptor/Method.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.3
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor;
+
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\Fqsen;
+use phpDocumentor\Reflection\Element;
+use phpDocumentor\Reflection\Php\Visibility;
+
+/**
+ * Descriptor representing a Method in a Class, Interface or Trait.
+ */
+final class Method implements Element
+{
+    /**
+     * @var DocBlock documentation of this method.
+     */
+    private $docBlock = null;
+
+    /**
+     * @var Fqsen Full Qualified Structural Element Name
+     */
+    private $fqsen;
+
+    /** @var bool $abstract */
+    private $abstract = false;
+
+    /** @var bool $final */
+    private $final = false;
+
+    /** @var bool $static */
+    private $static = false;
+
+    /** @var Visibility visibility of this method */
+    private $visibility = null;
+
+    /** @var Argument[] */
+    private $arguments = array();
+
+    /**
+     * Initializes the all properties.
+     * @param Fqsen $fqsen
+     * @param Visibility $visibility
+     * @param bool $abstract
+     * @param bool $static
+     * @param bool $final
+     */
+    public function __construct(
+        Fqsen $fqsen,
+        Visibility $visibility = null,
```

Comment:
Visibility is not optional, a method always has a visibility. If the visibility is omitted from a definition it is automatically public